x/nexus/keeper/keeper.go

```diff
@@ -68,6 +69,28 @@ func (k Keeper) GetParams(ctx sdk.Context) types.Params {
 	return p
 }
 
+// SetRouter sets the nexus router. It will panic if called more than once
+func (k *Keeper) SetRouter(router types.Router) {
+	if k.router != nil {
+		panic("router already set")
+	}
+
+	k.router = router
+
+	// In order to avoid invalid or non-deterministic behavior, we seal the router immediately
+	// to prevent additionals handlers from being registered after the keeper is initialized.
+	k.router.Seal()
+}
+
+// GetRouter returns the nexus router. If no router was set, it returns a (sealed) router with no handlers
+func (k Keeper) GetRouter() types.Router {
+	if k.router == nil {
+		k.SetRouter(types.NewRouter())
```

Comment:
Why set a default if not exist?

Reply:
So that we Seal the object. We could invoke `Seal` directly, but this way we avoid returning a nil reference.